### DISCUSSION TITLE
React flow state on adding new node isn't rendering newly added nodes (Using useContext Hook)

### DISCUSSION THREAD

**Discussion by shubh-am8** [xyflow/xyflow#2684] [General]
i am using useContext hook for data handling to fulfil my app's complex requirements, when i am adding new nodes using setNodes under useNodesState, only first node renders perfectly but after that my state variable has updated data but its not rendering on flow. can any one help me in this matter.

**Comment by amintabrizi** [comment]
It's late but I hope this sandbox helps people who have the same problem
https://codesandbox.io/s/unruffled-snowflake-qd5x22?file=/src/App.js
